### ISSUE TITLE
Dictionary assigns not working for API generated objects

### ISSUE THREAD

**Issue by darmagan** [Icinga/icinga2#5434]
## Problem
If for example a service is assigned to a custom dictionary of a host and that host or the attribute is created/modified over the API it will not match.

## Expected Behavior
The assign statement should match the host.

## Current Behavior
Statement is not matching the host

## Steps to Reproduce (for bugs)
1. modify/create object
```
curl -k -s -u root:root -H 'Accept: application/json' -X POST 'https://localhost:5665/v1/objects/hosts/box17.int.netways.de' -d '{"attrs": { "vars.mssql": { "service" : [ "MSSQLSERVER" ]  } } }' | python -m json.tool
```
2. Host vars look like this
```
...
"type": "Host",
                "vars": {
                    "mssql": {
                        "service": [
                            "MSSQLSERVER"
                        ]
                    },
                    "notification": {
                        "mail": {
                            "groups": [
                                "icingaadmins"
                            ]
                        }
                    },
...
```
3. Apply Service
```
apply Service "mssql_service" {
   import "generic-service"
   check_command = "dummy"
   assign where host.vars.mssql
}
```
4. Validate config
```
[2017-07-25 14:28:16 +0200] warning/ApplyRule: Apply rule 'mssql_service' (in /etc/icinga2/zones.d/master/hosts.conf: 1:0-1:28) for type 'Service' does not match anywhere!
```

## Your Environment
```
icinga2 - The Icinga 2 network monitoring daemon (version: r2.6.3-1)

Copyright (c) 2012-2017 Icinga Development Team (https://www.icinga.com/)
License GPLv2+: GNU GPL version 2 or later <http://gnu.org/licenses/gpl2.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.

Application information:
  Installation root: /usr
  Sysconf directory: /etc
  Run directory: /run
  Local state directory: /var
  Package data directory: /usr/share/icinga2
  State path: /var/lib/icinga2/icinga2.state
  Modified attributes path: /var/lib/icinga2/modified-attributes.conf
  Objects path: /var/cache/icinga2/icinga2.debug
  Vars path: /var/cache/icinga2/icinga2.vars
  PID path: /run/icinga2/icinga2.pid

System information:
  Platform: Ubuntu
  Platform version: 14.04.5 LTS, Trusty Tahr
  Kernel: Linux
  Kernel version: 3.13.0-101-generic
  Architecture: x86_64

Build information:
  Compiler: GNU 4.8.4
  Build host: lgw01-43
```

Regards

**Comment by Crunsher** [comment]
Related: #4084

**Comment by gunnarbeutner** [comment]
Closed because this is a duplicate of #4084 like @Crunsher mentioned already.
